keras_cv/models/segmentation/segment_anything/sam_mask_decoder.py

```diff
@@ -0,0 +1,230 @@
+        sparse_prompt_embeddings,
+        dense_prompt_embeddings,
+    ):
+        output_tokens = ops.concatenate(
+            [self.iou_token.weights[0], self.mask_tokens.weights[0]], axis=0
+        )
+        output_tokens = ops.broadcast_to(
+            output_tokens[None, ...],
+            shape=(
+                sparse_prompt_embeddings.shape[0],
+                output_tokens.shape[0],
+                output_tokens.shape[1],
+            ),
+        )
+        tokens = ops.concatenate(
+            [output_tokens, sparse_prompt_embeddings], axis=1
+        )
+
+        source = ops.broadcast_to(
+            image_embeddings,
+            shape=(
+                tokens.shape[0],
+                image_embeddings.shape[1],
+                image_embeddings.shape[2],
+                image_embeddings.shape[3],
+            ),
+        )
+        source = source + dense_prompt_embeddings
+        positional_source = ops.broadcast_to(
+            image_pe,
+            shape=(
+                tokens.shape[0],
+                image_embeddings.shape[1],
+                image_embeddings.shape[2],
+                image_embeddings.shape[3],
+            ),
+        )
+        B, H, W, C = source.shape
+
+        hidden_state, source = self.transformer(
+            source, positional_source, tokens
+        )
+        iou_token_out = hidden_state[:, 0, :]
+        mask_tokens_out = hidden_state[:, 1 : (1 + self.num_mask_tokens), :]
+
+        source = ops.reshape(source, (B, H, W, C))
+        upscaled_embeddings = self.output_upscaling(source)
+        hyper_in_list = []
+        for i in range(self.num_mask_tokens):
+            hyper_in_list.append(
+                self.output_hypernetworks_mlps[i](mask_tokens_out[:, i, :])
+            )
+        hyper_in = ops.stack(hyper_in_list, axis=1)
+        B, H, W, C = upscaled_embeddings.shape
+        upscaled_embeddings = ops.reshape(
+            ops.transpose(upscaled_embeddings, axes=(0, 3, 1, 2)),
+            (B, C, H * W),
+        )
+        masks = ops.reshape(
+            hyper_in @ upscaled_embeddings, (B, self.num_mask_tokens, H, W)
+        )
+
+        iou_pred = self.iou_prediction_head(iou_token_out)
+
+        return masks, iou_pred
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "transformer_dim": self.transformer_dim,
+                "transformer": keras.saving.serialize_keras_object(
+                    self.transformer
+                ),
+                "num_multimask_outputs": self.num_multimask_outputs,
+                "iou_head_depth": self.iou_head_depth,
+                "iou_head_hidden_dim": self.iou_head_hidden_dim,
+                "activation": self.activation,
+            }
+        )
+        return config
+
+    @classmethod
+    def from_config(cls, config):
+        config.update(
+            {"transformer": keras.layers.deserialize(config["transformer"])}
```

Comment:
Shouldn't we be able to pass the config here instead of the object if it's serializable?

Reply:
I don't think so. Keras errors out if I don't serialize and deserialize manually. RetinaNet and YOLOV8 also serialize and deserialize the layer arguments manually. Perhaps, @ianstenbit has a better idea about why this is required.